src/infoview.ts

```diff
@@ -167,13 +167,32 @@ export class InfoProvider implements TextDocumentContentProvider, Disposable {
 
     private updateMessages(): boolean {
         if (!this.curFileName) return false;
-        let msgs;
+        let msgs : Message[];
         switch (this.displayMode) {
         case DisplayMode.OnlyState:
+            /* Heuristic: find first position to the left which has messages attached,
+               from that on show all messages in this line */
             msgs = this.server.messages
                 .filter((m) => m.file_name === this.curFileName &&
-                    m.pos_line == this.curPosition.line + 1 &&
-                    m.pos_col == this.curPosition.character);
+                    m.pos_line == this.curPosition.line + 1)
+                .sort((a, b) => a.pos_col - b.pos_col);
+            let startColumn = undefined;
+            let startPos = null;
+            for (let i = 0; i < msgs.length; i++) {
+                if (this.curPosition.character < msgs[i].pos_col) break;
+                if (this.curPosition.character == msgs[i].pos_col) {
+                    startColumn = this.curPosition.character;
+                    startPos = i;
```

Comment:
`startPos` should be the **first** message with column `curPosition`. Note the `break` afterwards.